File: api/functionapp/order/Repositories/OrderRepository.cs
using System.Collections.Concurrent;
using OrderFunction.Models;

namespace OrderFunction.Repositories;

public class OrderRepository : IOrderRepository
{
    private readonly ConcurrentDictionary<string, Order> _orders = new();

    public Order Save(Order order)
    {
        _orders[order.Id] = order;
        return order;
    }

    public Order? FindById(string id)
    {
        _orders.TryGetValue(id, out var order);
        return order;
    }

    public List<Order> FindByUserId(string userId)
    {
        return _orders.Values
            .Where(o => o.UserId == userId)
            .ToList();
    }

    public long Count()
    {
        return _orders.Count;
    }
}

File: api/functionapp/order/src/main/java/com/example/repository/OrderRepository.java
package com.example.repository;

import com.example.model.Order;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Map;
import java.util.HashMap;
import java.util.Optional;
import java.util.stream.Collectors;

/**
 * In-memory repository for Order entities
 */
@Repository
public class OrderRepository {
    
    private final Map<String, Order> orders = new HashMap<>();
    
    /**
     * Save an order
     */
    public Order save(Order order) {
        orders.put(order.getId(), order);
        return order;
    }
    
    /**
     * Find order by ID
     */
    public Optional<Order> findById(String id) {
        return Optional.ofNullable(orders.get(id));
    }
    
    /**
     * Find all orders for a user
     */
    public List<Order> findByUserId(String userId) {
        return orders.values().stream()
                .filter(order -> order.getUserId().equals(userId))
                .collect(Collectors.toList());
    }
    
    /**
     * Get total order count
     */
    public long count() {
        return orders.size();
    }
}

